File: app/src/main/java/de/markusdamm/pokerapp/data/Player.java
package de.markusdamm.pokerapp.data;



public class Player implements Comparable<Player> {


    private boolean gender;
    private String name;
    private boolean selected, selectable;
    private int id;

    public String getName() {
        return name;
    }

    public Player(String name, int id, int gender){
        this.name = name;
        this.selected = false;
        this.id = id;
        this.selectable = true;
        this.gender = Gender.toBool(gender);
    }

    public void setSelectable(boolean selectable){
        this.selectable = selectable;
    }

    public void toggleSelected(){
        selected = !selected;
    }

    public boolean isSelectable(){
        return selectable;
    }

    public void setID(int id){
        this.id = id;
    }

    public int getId(){
        return id;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }

    public boolean isSelected() {
        return selected;
    }

    public String toString(){
        return name;
    }


    public void setGender(boolean gender){
        this.gender = gender;
    }

    public boolean getGender(){
        return gender;
    }

    public int getGenderAsInt(){
        return Gender.toInt(gender);
    }

    @Override
    public int compareTo(Player another) {
        return this.name.compareTo(another.getName());
    }
}

File: app/src/main/java/de/markusdamm/pokerapp/data/Player.kt
package de.markusdamm.pokerapp.data


class Player(val name: String, var id: Int, gender: Int) : Comparable<Player> {


    var gender: Boolean = false
    var isSelected: Boolean = false
    var isSelectable: Boolean = false

    val genderAsInt: Int
        get() = Gender.toInt(gender)

    init {
        this.isSelected = false
        this.isSelectable = true
        this.gender = Gender.toBool(gender)
    }

    fun toggleSelected() {
        isSelected = !isSelected
    }

    override fun toString(): String {
        return name
    }

    override fun compareTo(other: Player): Int {
        return this.name.compareTo(other.name)
    }
}